packages/types/lib/providers/provider.ts

```diff
@@ -135,4 +135,10 @@ export interface ProviderTwoStep extends BaseProvider {
     token_expires_in_ms?: number;
     proxy_header_authorization?: string;
 }
-export type Provider = BaseProvider | ProviderOAuth1 | ProviderOAuth2 | ProviderJwt | ProviderTwoStep;
+export interface ProviderSignatureBased extends BaseProvider {
+    signature_protocol: 'WSSE';
```

Comment:
With the new changes, we can now comfortably add support to more than one signature-based auth method, such as `WSSE` and `AWS_V4`, without needing to add a new auth mode. Thanks @TBonnin for suggesting this change.